p2p/p2p.go

```diff
@@ -35,6 +36,14 @@ const (
 	clientName                = "juno"
 )
 
+// Todo: this interface allows us to mock the P2P service until we implement additional tests / test infrastructure
+type WithBlockCh interface {
+	service.Service
+	WithBlockCh(blockCh chan p2pSync.BlockBody)
```

Comment:
```suggestion
	WithBlockCh(blockCh chan p2pSync.BlockBody)
```

Reply:
To do this, we can:
- Initialize the channel in `New`
- Return this channel in `Listen`
- Modify `pipeline.Bridge` utils to accept a channel as an argument instead of initializing it inside.